ddl/column.go

```diff
@@ -407,15 +416,36 @@ func onDropColumns(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		// write only -> delete only
 		job.SchemaState = model.StateDeleteOnly
 		setColumnsState(colInfos, model.StateDeleteOnly)
+		if len(idxInfos) > 0 {
+			setIndicesState(tblInfo, idxInfos, model.StateDeleteOnly)
+		}
 		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != colInfos[0].State)
 	case model.StateDeleteOnly:
 		// delete only -> reorganization
 		job.SchemaState = model.StateDeleteReorganization
 		setColumnsState(colInfos, model.StateDeleteReorganization)
+		if len(idxInfos) > 0 {
+			setIndicesState(tblInfo, idxInfos, model.StateDeleteReorganization)
+		}
 		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != colInfos[0].State)
 	case model.StateDeleteReorganization:
 		// reorganization -> absent
 		// All reorganization jobs are done, drop this column.
+		indexIDs := make([]int64, 0, len(idxInfos))
+		if len(idxInfos) > 0 {
+			newIndices := make([]*model.IndexInfo, 0, len(tblInfo.Indices))
+			for _, idx := range tblInfo.Indices {
+				if !indexInfoContains(idx.ID, idxInfos) {
+					newIndices = append(newIndices, idx)
+				}
+			}
+			tblInfo.Indices = newIndices
+			for _, indexInfo := range idxInfos {
+				indexIDs = append(indexIDs, indexInfo.ID)
+				dropIndexColumnFlag(tblInfo, indexInfo)
```

Comment:
Why dropIndexColumnFlag here? If the column is dropped finally, I don't think we need to set it anymore.

Reply:
Yes, you are right, no need to call this function anymore.